firmware/common2015/utils/logger/logger.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdarg>
+#include <string>
+
+#include "rj-macros.hpp"
+
+// Do weird macro things for logging the filename and line for every call.
+// Also allows for disabling all logging through macros so all log calls can be removed from production builds.
+#ifdef RJ_LOGGING_EN
+
+#ifdef __FILE_NAME__
+#define __BASE_FILE_NAME__  __FILE_NAME__
+#else   // __FILE_NAME__
+#define __BASE_FILE_NAME__  __FILE__
+#endif
+
+#define GEN_LOG_STR(x,y)    x ":" TO_STRING(y)
+#define LOG_LINE            GEN_LOG_STR(__BASE_FILE_NAME__, __LINE__)
+#define LOG(lvl, ...)       log(lvl, LOG_LINE, __func__, __VA_ARGS__)
+
+#else   // RJ_LOGGING_EN
+#define LOG(...)            // Nothing
+#endif
+
+
+#define FOREACH_LEVEL(LEVEL)    LEVEL(LOG_LEVEL_START)      \
```

Comment:
This is probably a lot more log levels than we'll need.  Unless we have a well-defined purpose for each one, want to just boil it down to INFO, WARN, and ERROR?